src/wayland-egldisplay.c

```diff
@@ -826,17 +825,12 @@ static bool getServerProtocolsInfo(struct wl_display *nativeDpy,
         /* use a second roundtrip to handle any wl_drm events triggered by binding the protocol */
         wl_display_roundtrip_queue(nativeDpy, queue);
 
-        if (!getDeviceFromDevIdInitialised) {
-            getDeviceFromDevId = dlsym(RTLD_DEFAULT, "drmGetDeviceFromDevId");
-            getDeviceFromDevIdInitialised = true;
-        }
-
         /*
          * if dmabuf feedback is available then use that. This will potentially
          * replace the drm_name provided by wl_drm, assuming the feedback provides
          * a valid dev_t.
          */
-        if (protocols->wlDmaBuf && getDeviceFromDevId) {
+        if (protocols->wlDmaBuf && protocols->pData->libdrmDlHandle && protocols->pData->getDeviceFromDevId) {
```

Comment:
Same as before, as long as we have `getDeviceFromDevId`, it doesn't matter if `libdrmDlHandle` is NULL.